src/e2e-test/java/io/cdap/plugin/gcs/stepsdesign/GCSPlugin.java

```diff
@@ -0,0 +1,512 @@
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.statusBanner, 5);
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 5);
+    wait.until(ExpectedConditions.invisibilityOf(CdfStudioLocators.statusBanner));
+  }
+
+  @Then("Preview and run the pipeline")
+  public void previewAndRunThePipeline() {
+    SeleniumHelper.waitAndClick(CdfStudioLocators.preview, 5L);
+    CdfStudioLocators.runButton.click();
+  }
+
+  @Then("Verify the preview of pipeline is {string}")
+  public void verifyThePreviewOfPipelineIs(String previewStatus) {
+    WebDriverWait wait = new WebDriverWait(SeleniumDriver.getDriver(), 180);
+    wait.until(ExpectedConditions.visibilityOf(CdfStudioLocators.statusBanner));
+    Assert.assertTrue("Preview of pipeline should " + previewStatus,
+                      CdfStudioLocators.statusBannerText.getText().contains(previewStatus));
+    if (!previewStatus.equalsIgnoreCase("failed")) {
+      wait.until(ExpectedConditions.invisibilityOf(CdfStudioLocators.statusBanner));
+    }
+  }
+
+  @Then("Click on preview data for GCS")
+  public void clickOnPreviewDataForGCS() {
+    openSinkPluginPreviewData("GCS");
+  }
+
+  @Then("Click on preview data for BigQuery")
+  public void clickOnPreviewDataForBigQuery() {
+    openSinkPluginPreviewData("BigQuery");
+  }
+
+  @Then("Verify preview output schema matches the outputSchema captured in properties")
+  public void verifyPreviewOutputSchemaMatchesTheOutputSchemaCapturedInProperties() {
+    List<String> previewSchemaColumnList = new ArrayList<>();
+    for (WebElement element : CdfStudioLocators.previewInputRecordColumnNames) {
+      previewSchemaColumnList.add(element.getAttribute("title"));
+    }
+    Assert.assertTrue("Schema column list should be equal to preview column list",
+                      previewSchemaColumnList.equals(propertiesSchemaColumnList));
+    CdfStudioLocators.previewPropertiesTab.click();
+    Map<String, String> previewSinkInputSchema = new HashMap<>();
+    int index = 0;
+    for (WebElement element : CdfSchemaLocators.inputSchemaColumnNames) {
+      previewSinkInputSchema.put(element.getAttribute("value"),
+                                 CdfSchemaLocators.inputSchemaDataTypes.get(index).getAttribute("title"));
+      index++;
+    }
+    Assert.assertTrue("Schema should match", previewSinkInputSchema.equals(sourcePropertiesOutputSchema));
+  }
+
+  @Then("Close the preview")
+  public void closeThePreview() {
+    CdfStudioActions.previewSelect();
+  }
+
+  @Then("Close the preview data")
+  public void closeThePreviewData() {
+    CdfStudioActions.clickCloseButton();
+    CdfStudioActions.previewSelect();
+  }
+
+  @Then("Deploy the pipeline")
+  public void deployThePipeline() {
+    SeleniumHelper.waitElementIsVisible(CdfStudioLocators.pipelineDeploy, 2);
+    CdfStudioActions.pipelineDeploy();
+  }
+
+  @Then("Save and Deploy Pipeline")
+  public void saveAndDeployPipeline() throws InterruptedException {
```

Comment:
Can we move these common step definitions which will be required in scenarios related to every plugin move to some common class?

Reply:
This comment applies to all common steps related to pipeline run like `Deploy the pipeline`, `Save and Deploy Pipeline`, `Run the pipeline`, `Wait till pipeline is in running state`, etc and all the steps related to preview run. So that they can be re-used.

Reply:
Agreed. We can create class in e2e-framework for common step definitions.

Reply:
Raised https://github.com/cdapio/cdap-e2e-tests/pull/63. Please review.